packages/customize-widgets/src/components/sidebar-block-editor/use-sidebar-block-editor.js

```diff
@@ -209,7 +209,8 @@ export default function useSidebarBlockEditor( sidebar ) {
 					}
 				} else {
 					const widget = blockToWidget( nextBlock );
-					sidebar.addWidget( widget, index );
+					const widgetId = sidebar.addWidget( widget, index );
+					nextBlock.attributes.__internalWidgetId = widgetId;
```

Comment:
Does mutating `nextBlock` here mean that we're setting `__internalWidgetId` on the block that's in the `@wordpress/block-editor` store's state? Not sure if that has unexpected repercussions. It might be safer to make a copy of the block with `{ ...nextBlock }`.